### ISSUE TITLE
[linux55-tkg] Navi 10 and iwlwifi patches?

### ISSUE THREAD

**Issue by gardotd426** [Tk-Glitch/PKGBUILDS#479]
Hey Etienne, so when I was using your linux55-rc-tkg PKGBUILD I was having to add in the navi 10 overclocking patches, and I still had to use them at first when regular linux55-tkg came out. And since linux55-tkg came out and it has the iwlwifi bug where certain iwlwifi devices (including mine) don't work on linux-5.5, I also added my iwlwifi patch. Well, as of about a week ago, I can't get it to build with those patches in the same directory as the PKGBUILD. No matter what I do (delete the linux source .xz tarball, delete the patch directory, try a fresh `git clone`, etc), it gives me the same error, and says previously applied or reversed patch detected. It would also say something like:
```
1 out of 1 hunk ignored -- saving rejects to file drivers/gpu/drm/amd/powerplay/navi10_ppt.c.rej
```
so I'd try to find `navi10_ppt.c.rej` to find where the source code was being kept, and it's nowhere to be found. I tried `sudo updatedb` followed by `locate navi10_ppt.c.rej`, searching for it within nemo and pcmanfm, it was nowhere on the system. It's as if the source code with the rejected patched files gets deleted (maybe by the "exit cleanup"?) 

So, I made a new directory in my `~/gits/` dir, `~/gits/tkg/~`, cd'd into it and cloned a fresh git so I could start brand new. I copied over my other `customization.cfg`, and the `*.mypatch` files, and literally nothing else. I also looked in `../community-patches/linux55-tkg` and read its `README.md` to make sure you hadn't maybe added it. Nothing there, but literally right now as I was typing this I was like "hmm, let me check one more thing," and I ran `grep -i navi linux55-tkg/linux55-tkg-patches/*` and sure enough, I found results for that (and `grep -i iwlwifi`) in `0003-glitched-base.patch` (duh). 

So I THINK that answers my question, and I don't want to waste your time, but I just want to make sure which patches are or aren't applied. Are all 3 navi overclocking patches from https://gitlab.freedesktop.org/drm/amd/issues/1020#note_394351, along with the iwlwifi 5.5 fix included? I'm also assuming that [this patch](https://gitlab.freedesktop.org/drm/amd/issues/1020#note_400803) for the voltages which was just posted today on that same thread is NOT included? 

So basically, these are the patches I think are included:
```
1-3-drm-amdgpu-navi-fix-for-OD-MCLK.mypatch
2-3-drm-amdgpu-navi10-add-OD_range-for-oc.mypatch
3-3-drm-amdgpu-navi10-add-support-for-restoring-default-table.mypatch
```
PLUS the iwlifi patch, but NOT this:
```
drm-amdgpu-fetch-default-VDDC-curve-voltages.mypatch
```

Thanks for your help and/or clarification

EDIT: forgot a parenthesis

**Comment by Tk-Glitch** [comment]
You are correct :)

**Comment by gardotd426** [comment]
I figured it out for myself for once?!?!?! Shocker. Lol but seriously, thank you, and am I also correct in my statement about the `.rej` files being deleted on cleanup? It's not really an issue if so, but I just wanted to make sure so I don't go on wild goose chases in the future for a file that doesn't exist.

**Comment by Tk-Glitch** [comment]
Yup, you nailed it 😃 . You can prevent the clearing step from your .cfg if needed:
```
# Set to anything else than "true" to limit cleanup operations and keep source and files generated during compilation.
# Default is "true".
_NUKR="true"
```

**Comment by gardotd426** [comment]
Okay awesomeness. Might I suggest, if it's not too much bother, adding the new patch that makes the default voltages on `VDDC_CURVE` work? I've tested it applied to your latest and it works like a charm. Version 2 is [here](https://gitlab.freedesktop.org/drm/amd/issues/1020#note_400803).

**Comment by gardotd426** [comment]
I'll go ahead and close this all the way. For anyone else that wants to know:
- iwlwifi 5.5 bugfix patch is already included
- The 3 Navi overclocking patches not including the default voltage values patch are included as of 4/2/20

**Comment by Tk-Glitch** [comment]
Yes I'm indeed planning to add the default navi vddc curve values patch in the mix. Likely later today. As a side note, keep in mind that setting `_NUKR` to `false` (or anything not `true`) will prevent subsequent building as you'll get a dirty tree in the way. It's a rough debugging feature in its current form 🐸

**Comment by gardotd426** [comment]
Okay awesome, thanks. Yeah I don't plan to use it unless a situation like
this arises where I need to figure out why a patch ain't workin'. But
thanks man! Keep up the great work, you're a gem.

**Comment by Tk-Glitch** [comment]
I'm touched. Thank you very much!

**Comment by Tk-Glitch** [comment]
btw, this is for my 5700XT but should apply to any currently available Navi GPU :
```
set -e
set -x
cd /sys/class/drm/card0/device # change path to your Navi GPU if needed
pushd hwmon/hwmon*
echo 220000000 | sudo tee power1_cap # Max GPU power allowed (here set to 220W)
popd
echo 's 0 800' > pp_od_clk_voltage # Min GPU clock, capped at 800 min without custom ppt
echo 's 1 2070' > pp_od_clk_voltage # Max GPU clock, capped at 2150 max without custom ppt
echo 'm 1 980' > pp_od_clk_voltage # Memory clock - currently static on OD but still moves around - capped at 950 max without custom ppt - This is wrongly listed as state 0 on kernel 5.5
echo 'vc 0 800 750' > pp_od_clk_voltage # lower end of the curve, state 0, 800MHz GPU clock for 750mV
echo 'vc 1 1400 950' > pp_od_clk_voltage # mid, state 1, 1400MHz GPU clock for 950mV
echo 'vc 2 2040 1100' > pp_od_clk_voltage # high, state 2, 2040MHz GPU clock for 1100mV
echo 'c' > pp_od_clk_voltage # Validating settings to apply
```
On Navi the clocks will move around as you get power/temperature limited, but if you lock the "Max GPU clock" to a freq that will never hit temp/power thresholds, you'll effectively get a static max freq.
